### PR TITLE
Person student and teacher classes

### PR DESCRIPTION
**Pull request by clin2on3mun**
in this milestone:
- created Person Class
- Created Teacher and student as sub classes and Inherit Person class

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (Olamarx) [inline person.rb:19]: - [ ] I will encourage you to make this method private by adding private to the line before the method as required by the project requirement.
image

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(classroom, age, name = 'unknown')
```
Comment (Olamarx) [inline student.rb]: - [ ] This constructor is meant to have all the parameters of Person class too and by inclusion, classroom. I will encourage you to add the parent's permission too.

image


Remember to include it in the super too.

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(age, specialization, name = 'unknown')
+    super(name, age)
+    @specialization = specialization
+  end
+
```
Comment (Olamarx) [inline teacher.rb]: - [ ] Same thing as recommended in the student class applies to this.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @clin2on3mun,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- All methods in place ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @clin2on3mun 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Classes added properly 👍 
* Good use of ruby syntax 👍 
* No linter errors 👍 
* Well written PR 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**